master/internal/db/postgres_checkpoints.go

```diff
@@ -0,0 +1,86 @@
+package db
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+// FilterForRegisteredCheckpoints gets the deleted checkpoints provided in the model registry.
+func (db *PgDB) FilterForRegisteredCheckpoints(deleteCheckpoints []uuid.UUID) ([]uuid.UUID, error) {
+	var checkpointIDRows []struct {
+		ID uuid.UUID
+	}
+
+	if err := db.queryRows(`
+	SELECT DISTINCT(c.uuid) AS ID FROM checkpoints_view AS c 
+	JOIN model_versions AS mv ON mv.checkpoint_uuid = c.uuid
+	WHERE c.uuid IN (SELECT UNNEST($1::uuid[])); 
+`, &checkpointIDRows, deleteCheckpoints); err != nil {
+		return nil, fmt.Errorf(
+			"querying for all requested delete checkpoints registered in model registry: %w", err)
```

Comment:
the error is describing the callers context - let them describe that.
`fmt.Errorf("filtering checkpoint uuids (%v) by those registered in the model registry: %w", uuids, err)`